### PR TITLE
Facepile: New FacepileButton and other styling fixes for the component and its documentation

### PR DESCRIPTION
**Pull request by lynamemi**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

- The BaseButton used in Facepile had no styling so the screenReader text was visible on the page, making the descriptive overflow button super wonky. A new FacepileButton replaces this. Also updated the jest snapshot to account for this change.
- Reordered some of the controls + their spacing in examples to be more clearly organized and documented
- Removed the 'Additional Data' dropdown because it wasn't doing anything anymore, and I don't think Facepile has that functionality now - correct me if I'm missing a use case
- Added new color to Persona to allow for styling of the overflowButton to match the toolkit

# **Before:**
facepile-before

# **After:**
facepile-after

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by lynamemi** [comment]
The screener change is intended!

File: packages/office-ui-fabric-react/src/components/Persona/Persona.scss [hunk 1]
```diff
@@ -9,21 +9,22 @@
...
-$ms-color-initials-darkGreen:   #1E7145;
-$ms-color-initials-lightPink:   #E773BD;
-$ms-color-initials-pink:        #FF0097;
-$ms-color-initials-magenta:     #7E3878;
-$ms-color-initials-purple:      #603CBA;
-$ms-color-initials-black:       #1D1D1D;
-$ms-color-initials-orange:      #DA532C;
-$ms-color-initials-red:         #EE1111;
-$ms-color-initials-darkRed:     #B91D47;
+$ms-color-initials-overflowButton:   #EAEAEA;
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.scss]: I'm not seeing this color get used

File: packages/office-ui-fabric-react/src/components/Persona/Persona.Props.ts [hunk 2]
```diff
@@ -167,5 +167,6 @@ export enum PersonaInitialsColor {
   black = 11,
   orange = 12,
   red = 13,
-  darkRed = 14
+  darkRed = 14,
+  overflowButton = 15
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.Props.ts]: `overflowButton` color isn't defined here.

https://github.com/OfficeDev/office-ui-fabric-react/blob/7f4ece990b065b1d43ec71cd4d4615b5ede0d082/packages/office-ui-fabric-react/src/components/Persona/PersonaConsts.ts#L30

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.tsx [hunk 3]
```diff
@@ -180,40 +181,43 @@ export class Facepile extends BaseComponent<IFacepileProps, {}> {
           title={ personaNames }
           size={ personaSize }
           onRenderInitials={ this._renderInitialsNotPictured(numPersonasNotPictured) }
+          initialsColor={ PersonaInitialsColor.overflowButton }
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.tsx]: The color for `PersonaInitialsColor.overflowButton` is not defined (see other comments below). Is it supposed to be `$ms-color-initials-overflowButton:   #EAEAEA;`? The bg-color of `ms-Facepile-descriptiveOverflowButton` is also `#EAEAEA`

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.tsx [hunk 4]
```diff
@@ -180,40 +181,43 @@ export class Facepile extends BaseComponent<IFacepileProps, {}> {
...
-      <BaseButton
+      <FacepileButton
         {...overflowButtonProps}
         className={ css('ms-Facepile-overflowButton', 'ms-Facepile-itemButton', styles.overflowButton, styles.itemButton) }
       >
         <PersonaCoin
           size={ personaSize }
-          onRenderInitials={ this._renderInitials(icon) }
+          onRenderInitials={ this._renderInitials(icon, overflowInitialsIcon) }
+          initialsColor={ PersonaInitialsColor.overflowButton }
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.tsx]: Same here. `ms-Facepile-overflowButton` is also `#EAEAEA`

File: packages/office-ui-fabric-react/src/components/Persona/Persona.Props.ts (hunk 2 above)
Comment (lynamemi) [reply packages/office-ui-fabric-react/src/components/Persona/Persona.Props.ts]: this const is the missing piece I was looking for, thanks! I'll take a look at how this piece fits in the puzzle.

**Comment by lynamemi** [comment]
@erichdev I figured out why the `PersonaInitialsColor `was sort of working, but not really. The `initialsColor={ PersonaInitialsColor.lightGrey }` prop was need to override the default blue bg-color, but when the compiler didn't find a color defined in PersonaConst, it defaulted to the `.overflowButton` class, which sets the bg to EAEAEA, the same as the intended `PersonaInitialsColor `color. 

With the latest commit, I explicitly defined the PersonaConst lightGrey color. Instead of calling the enum overflowButton to define its intended use, I followed the method used in [this PR](https://github.com/OfficeDev/office-ui-fabric-react/pull/3249) to define lightGrey as a reserved initials color.

Even though the sass bg color is not being used, I decided to leave it for now since it's acting as a fall back default color, overriding the blue. I could also see removing that though, as it's a bit confusing.

Calling on @mtennoe for input for reserving lightGrey.

**Comment by lynamemi** [comment]
Seems that the screener change is coming from https://github.com/OfficeDev/office-ui-fabric-react/pull/3249

Should be copacetic @betrue-final-final

File: packages/office-ui-fabric-react/src/components/Persona/Persona.scss [hunk 5]
```diff
@@ -9,21 +9,22 @@
...
-$ms-color-initials-darkGreen:   #1E7145;
-$ms-color-initials-lightPink:   #E773BD;
-$ms-color-initials-pink:        #FF0097;
-$ms-color-initials-magenta:     #7E3878;
-$ms-color-initials-purple:      #603CBA;
-$ms-color-initials-black:       #1D1D1D;
-$ms-color-initials-orange:      #DA532C;
-$ms-color-initials-red:         #EE1111;
-$ms-color-initials-darkRed:     #B91D47;
+$ms-color-initials-lightGrey:        #EAEAEA;
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.scss]: I believe EAEAEA is defined somewhere else as a SASS variable, can you use that variable here to reduce duplication?

File: packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx [hunk 6]
```diff
@@ -34,7 +34,9 @@ const SIZE_TO_PIXELS = {
 };
 
 /**
- * Red is a color that often has a special meaning, so it is considered a reserved color and can only be set with overrides
+ * These colors are considered reserved colors and can only be set with overrides:
+ * Red is a color that often has a special meaning.
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Persona/PersonaCoin.tsx]: Please add bullets/dots/numbering for red and lightgrey, to make it clearer that this is a list of colors :)

**Comment by mtennoe** [comment]
Adding lightGrey as an override color is unproblematic, as long as it is not added to the palette (due to accessibility reasons, like you mention in your comments).

Just a thought: Would it work to make the color transparent instead, to make it more generic?

**Comment by lynamemi** [comment]
@mtennoe I like that idea of adding a transparent color instead of light grey. I could definitely see flexibility coming in handy in the future for different use cases, and it works smoothly with the sass classes we already have in place for Facepile.

File: packages/office-ui-fabric-react/src/components/Facepile/FacepileButton.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { BaseComponent } from '../../Utilities';
+import { BaseButton, IButtonProps } from '../../Button';
+import { getTheme } from '../../Styling';
+import { getStyles } from '../Button/BaseButton.styles';
+
+export class FacepileButton extends BaseComponent<IButtonProps, {}> {
+
+  public render() {
+    let baseButtonStyles = getStyles(getTheme());
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/Facepile/FacepileButton.tsx]: baseButtonStyles doesnt change, so can be a const

**Pull request merged**